demo/src/__tests__/MemoryModelsUserInput.spec.tsx

```diff
@@ -174,4 +213,71 @@ describe("MemoryModelsUserInput", () => {
             });
         });
     });
+
+    describe("MemoryModelsConfigInput", () => {
+        beforeEach(() => {
+            render(
+                <MemoryModelsUserInput
+                    onTextDataSubmit={onSubmitMock}
+                    setTextData={setTextDataMock}
+                    textData={textDataMock}
+                    setFailureBanner={setFailureBannerMock}
+                    jsonResult={jsonResult}
+                    configData={configDataMock}
+                    setConfigData={setConfigDataMock}
+                />
+            );
+        });
+
+        it("renders a number input with correct props and checkbox that is checked by default", () => {
+            const seedInput: HTMLInputElement =
+                screen.getByTestId("config-seed");
+            //TODO: get max and min values from backend config
+            [
+                ["max", (2 ** 31).toString()],
+                ["min", "0"],
+                ["type", "number"],
+            ].forEach(([property, value]) => {
+                expect(seedInput).toHaveProperty(property, value);
+            });
+
+            // According to https://mui.com/material-ui/react-checkbox/#accessibility, checkboxes should always
+            // have labels, so testing with label text rather than data-testid here.
+            const automationCheckbox: HTMLInputElement = screen.getByLabelText(
+                "Generate with automation"
+            );
+            expect(automationCheckbox.checked).toBe(true);
+        });
+
+        it("handles seed change", () => {
+            const seedInput: HTMLInputElement =
+                screen.getByTestId("config-seed");
+            const mockSeed = "123";
+            fireEvent.change(seedInput, { target: { value: mockSeed } });
+            expect(setConfigDataMock).toHaveBeenNthCalledWith(
+                1,
+                expect.objectContaining({
```

Comment:
I'm not sure why you're using `objectContaining` instead of `toEqual` here (and elsewhere)?

Reply:
Good catch, used `objectContaining` initially when the order wasn't preserved, fixed